eng/Versions.props

```diff
@@ -82,5 +82,19 @@
     Versions below this comment are not managed by automation and can be changed as needed.
   -->
   <PropertyGroup Label="Manual">
+    <!-- Compatibility with VS 17.4/.NET SDK 7.0.1xx -->
+    <MicrosoftCodeAnalysisVersion_4_4>4.4.0</MicrosoftCodeAnalysisVersion_4_4>
+    <!-- Compatibility with VS 17.8/.NET SDK 8.0.1xx -->
+    <MicrosoftCodeAnalysisVersion_4_8>4.8.0</MicrosoftCodeAnalysisVersion_4_8>
+    <!-- Compatibility with the latest Visual Studio Preview release -->
+    <!--
+      The exact version is always a moving target. This version should never go ahead of the version of Roslyn that is included in the most recent
+      public Visual Studio preview version. If it were to go ahead, then any components depending on this version would not work in Visual Studio
+      and would cause a major regression for any local development that depends on those components contributing to the build.
+      This version must also not go ahead of the most recently release .NET SDK version, as that would break the source-build build.
+      Source-build builds the product with the most recent previously source-built release. Thankfully, these two requirements line up nicely
+      such that any version that satisfies the VS version requirement will also satisfy the .NET SDK version requirement because of how we ship.
+    -->
+    <MicrosoftCodeAnalysisVersion_LatestVS>4.8.0</MicrosoftCodeAnalysisVersion_LatestVS>
```

Comment:
This looks like a copy from https://github.com/dotnet/runtime/blob/59c2ea578bd615a63d56e8ff4b1de0a6b824691f/src/coreclr/tools/Directory.Build.props#L16-L25, though with a different value.

Overall, this property isn't being explicitly used. @dariusclay, is it needed at all? If it does - we need comments explaining why and how it is used. If not - it shouldn't be here.

Reply:
It's from https://github.com/dotnet/runtime/blob/dc0432bcb6e7afba54c426f15d03fe8c8fd0d0aa/eng/Versions.props, see comment above

Reply:
Since we're not using this, it should be removed.

Reply:
Cleaned this part up.